aws-rds-customdbengineversion/src/main/java/software/amazon/rds/customdbengineversion/Translator.java

```diff
@@ -0,0 +1,120 @@
+package software.amazon.rds.customdbengineversion;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import software.amazon.awssdk.services.rds.model.CreateCustomDbEngineVersionRequest;
+import software.amazon.awssdk.services.rds.model.DBEngineVersion;
+import software.amazon.awssdk.services.rds.model.DeleteCustomDbEngineVersionRequest;
+import software.amazon.awssdk.services.rds.model.DescribeDbEngineVersionsRequest;
+import software.amazon.awssdk.services.rds.model.ModifyCustomDbEngineVersionRequest;
+import software.amazon.rds.common.handler.Tagging;
+
+public class Translator {
+    public static CreateCustomDbEngineVersionRequest createCustomDbEngineVersionRequest(
+            final ResourceModel model,
+            final Tagging.TagSet tags
+    ) {
+        return CreateCustomDbEngineVersionRequest.builder()
+                .databaseInstallationFilesS3BucketName(model.getDatabaseInstallationFilesS3BucketName())
+                .databaseInstallationFilesS3Prefix(model.getDatabaseInstallationFilesS3Prefix())
+                .description(model.getDescription())
+                .engine(model.getEngine())
+                .engineVersion(model.getEngineVersion())
+                .kmsKeyId(model.getKMSKeyId())
+                .manifest(model.getManifest())
+                .tags(Tagging.translateTagsToSdk(tags))
+                .build();
+    }
+
+    static DescribeDbEngineVersionsRequest describeDbEngineVersionsRequest(final ResourceModel model) {
+        return DescribeDbEngineVersionsRequest.builder()
+                .engine(model.getEngine())
+                .engineVersion(model.getEngineVersion())
+                .build();
+    }
+
+    static ModifyCustomDbEngineVersionRequest modifyCustomDbEngineVersionRequest(final ResourceModel model) {
+        return ModifyCustomDbEngineVersionRequest.builder()
+                .description(model.getDescription())
+                .engine(model.getEngine())
+                .engineVersion(model.getEngineVersion())
+                .status(model.getStatus())
+                .build();
+    }
```

Comment:
It is hard to spot it right away, but could you please check if any of these parameters should be omitted upon an update? (aka: shall we use diff anywhere here?)

Reply:
Btw, Is it being applied immediately?